recipes/harfbuzz/all/conanfile.py

```diff
@@ -92,48 +95,81 @@ def requirements(self):
         if self.options.with_icu:
             self.requires("icu/71.1")
         if self.options.with_glib:
-            self.requires("glib/2.73.3")
+            self.requires("glib/2.74.1")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def generate(self):
+        def is_enabled(value):
+            return "enabled" if value else "disabled"
+
+        def meson_backend_and_flags():
+            if is_msvc(self) and Version(self.settings.compiler.version) == "15" and self.settings.build_type == "Debug":
+                # Mitigate https://learn.microsoft.com/en-us/cpp/build/reference/zf?view=msvc-170
+                return "vs", ["/bigobj"]
+            return "ninja", []
+
+        PkgConfigDeps(self).generate()
+
+        backend, cxxflags = meson_backend_and_flags()
+        tc = MesonToolchain(self, backend=backend)
+        tc.project_options.update({
+            "glib": is_enabled(self.options.with_glib),
+            "icu": is_enabled(self.options.with_icu),
+            "freetype": is_enabled(self.options.with_freetype),
+            "gdi": is_enabled(self.options.get_safe("with_gdi")),
+            "directwrite": is_enabled(self.options.get_safe("with_directwrite")),
+            "gobject": is_enabled(can_run(self)),
+            "tests": "disabled",
+            "docs": "disabled",
+            "benchmark": "disabled",
+            "icu_builtin": "false"
+        })
+        tc.cpp_args += cxxflags
```

Comment:
Is this variable documented for `MesonToolchain`?

Reply:
Yes, although the documentation isn't all that clear for the `cpp_*` attributes.

![Screenshot from 2022-11-11 16-33-30](https://user-images.githubusercontent.com/422117/201386756-6ddb1e83-5951-4e29-aa7e-a1efad70bfeb.png)

Reply:
Hmm, I don't see any recipes using this yet. @uilianries Is it safe to use `cpp_args` in Meson toolchain? It seems like the best choice here, but is only mentioned in a bit of an off-hand way in the documentation.